### PR TITLE
Add `RemoteAddress` to mirrors

### PR DESCRIPTION
**Pull request by KN4CK3R**
This PR adds a new field `RemoteAddress` to both mirror types which contains the sanitized remote address for easier (database) access to that information. Will be used in the audit PR if merged.

### CODE REVIEW

**Comment by puni9869** [comment]
Any screenshots, if applicable?

File: modules/util/url.go [hunk 1]
```diff
@@ -39,3 +39,12 @@ func URLJoin(base string, elems ...string) string {
 	}
 	return joinedURL
 }
+
+func SanitizeURL(s string) (string, error) {
+	u, err := url.Parse(s)
```
Comment (puni9869) [inline modules/util/url.go:44]: Make sure url is `urlDecoded`.

File: routers/web/repo/view.go [hunk 2]
```diff
@@ -659,7 +650,7 @@ func checkHomeCodeViewable(ctx *context.Context) {
 
 			ctx.Data["Repo"] = ctx.Repo
 			ctx.Data["MigrateTask"] = task
-			ctx.Data["CloneAddr"] = safeURL(cfg.CloneAddr)
+			ctx.Data["CloneAddr"], _ = util.SanitizeURL(cfg.CloneAddr)
```
Comment (puni9869) [inline routers/web/repo/view.go:653]:
```suggestion
			ctx.Data["CloneAddr"], _ = util.SanitizeURL(cfg.CloneAddr)
```
Any thoughts why we are suppressing the error aka `_`

File: modules/util/url.go [hunk 3]
```diff
@@ -39,3 +39,12 @@ func URLJoin(base string, elems ...string) string {
...
+	if err != nil {
+		return "", err
```
Comment (puni9869) [inline modules/util/url.go:46]: 👍

**Review by silverwind — APPROVED** [review]

File: modules/util/url.go (hunk 1 above)
Comment (KN4CK3R) [reply modules/util/url.go:44]: What do you mean? (For this PR) the url was already parsed somewhere else already.

**Review by puni9869 — APPROVED** [review]

File: models/migrations/v1_21/v276.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := sess.Iterate(new(Mirror), func(_ int, bean any) error {
+		m := bean.(*Mirror)
```
Comment (delvh) [inline models/migrations/v1_21/v276.go:54]:
```suggestion
		m := bean.(*Mirror)
		if m.RemoteAddress != "" {
			return nil
		}
```

File: models/migrations/v1_21/v276.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := sess.Iterate(new(PushMirror), func(_ int, bean any) error {
+		m := bean.(*PushMirror)
```
Comment (delvh) [inline models/migrations/v1_21/v276.go:87]:
```suggestion
		m := bean.(*PushMirror)
		if m.RemoteAddress != "" {
			return nil
		}
```

**Pull request merged**